Create a backup of Kraken configs

gochist opened the pull request:
This will make backup of kraken config file in cluster directory when the config file is changed.

```
june@hp650:~/.kraken/june$ ls -al
total 60
drwxrwxr-x 3 june june 4096 Dec 12 10:43 .
drwxrwxr-x 3 june june 4096 Dec 12 10:43 ..
-rw------- 1 june june 5642 Dec 12 10:17 admin.kubeconfig
-rw-r--r-- 1 june june   61 Dec 12 10:43 cluster.status.lock
-rw-rw-r-- 1 june june 6254 Dec 12 10:43 config.yaml
-rw-rw-r-- 1 june june 6254 Dec 12 10:34 config.yaml.21932.2017-12-12@10:37:31~
-rw-rw-r-- 1 june june 6253 Dec 12 10:37 config.yaml.23572.2017-12-12@10:40:01~
-rw-rw-r-- 1 june june 6257 Dec 12 10:40 config.yaml.26739.2017-12-12@10:43:11~
-rw-rw-r-- 1 june june 1767 Dec 12 10:13 deploymentmanager-create.yaml
drwxrwxr-x 5 june june 4096 Dec 12 10:17 .helm
-rw-rw-r-- 1 june june    0 Dec 12 10:17 ssh_config
```

davidewatson commented on ansible/roles/kraken.config/tasks/main.yaml:
```diff
@@ -31,6 +31,25 @@
+  with_items:
+    - "{{ kraken_config.clusters }}"
+  loop_control:
+    loop_var: cluster
```
If we save only the current and previous config, then if an operation fails and is rerun, won't the current and previous config be the same? Ah no, since you propose hashing the file contents.

IMHO, this sort of complexity belongs in an ansible module written in python.

gochist replied:
Or... We can let the ansible makes backup and then cleanup the old config file.

gochist commented on ansible/roles/kraken.config/tasks/main.yaml:
```diff
@@ -31,6 +31,25 @@
+  with_items:
+    - "{{ kraken_config.clusters }}"
+  loop_control:
+    loop_var: cluster
```
@coffeepac I found that it doesn't work for GKE cluster, it should be repeated for GKE.

gochist commented:
- I tried to move it from kraken.config to kraken.cluster_common. But I realized that GKE don't use cluster_common. So I left those tasks in kraken.config. 
- added `backup-config.yaml` to avoid repeated loops.
- I added backup retention.

coffeepac commented:
@gochist this PR is failing CI.  specifically its failing the Ansible linter, the find line is too long.  you'll need to break it into two lines.

gochist commented:
@coffeepac To pass ansible-lint, adding handler was needed.

leahnp commented:
Should we rebase this to run it through gitlab ci?

coffeepac commented:
@leahnp I'm fine with merging this without going through gitlab CI.  it passed the old required check so I'm going to merge it.
